### ISSUE TITLE
Problems with network architecture, may lead to high overfitting

### ISSUE THREAD

**Issue by Ao-Lee** [davidsandberg/facenet#631]
the network uses two fully connected layer in the end. we assume inception_res_v1 with center loss is used:

the first layer is used to get the 128 dementional embadding presentation of each face image. It is implemented in inception_resnet_v1.py as the following:

`net = slim.fully_connected(net, bottleneck_layer_size, activation_fn=None, scope='Bottleneck', reuse=False)
`
the second layer is used to do classification  task to generate softmax loss (also center loss). It is implemented in train_tripletloss.py.py as the following:

```
prelogits, _ = network.inference(image_batch, args.keep_probability, 
            phase_train=phase_train_placeholder, bottleneck_layer_size=args.embedding_size, 
            weight_decay=args.weight_decay)

logits = slim.fully_connected(prelogits, len(train_set), activation_fn=None, 
                weights_initializer=tf.truncated_normal_initializer(stddev=0.1), 
                weights_regularizer=slim.l2_regularizer(args.weight_decay),
                scope='Logits', reuse=False)
```
It seems that there is no non-linearity function in between and there is only a batch-norm layer between the two fc layers, which means that the two fc layers act as only one linear transformation. 
Consider two linear layers without biases, y= Ax and t = By, if we stack the two layers togather, we got t = (BA)x, this is a simple linear transformation.

But I wonder why the model performs so well on LFW (I have downloaded and tested the model, with over 98% accuracy, modifing align function is needed to achive that ).

**Comment by davidsandberg** [comment]
Closing due to inactivity. Reopen if needed.

**Comment by Edwardmark** [comment]
Why not use relu in the first fc layer?I donnot understand.
